src/Npgsql/ConnectorPool.cs

```diff
@@ -264,8 +264,10 @@ bool CheckIdleConnector([NotNullWhen(true)] NpgsqlConnector? connector)
                 if (i == _max)
                     throw new NpgsqlException($"Could not find free slot in {Connectors} when opening. Please report a bug.");
 
-                // Only start pruning if it was this thread that incremented open count past _min.
-                if (numConnectors == _min)
+                // Only start pruning if we've incremented open count past _min.
+                // Note that we don't do it only once because the thread which incremented open count past _min might get exception
```

Comment:
```suggestion
                // Note that we don't do it only once, on equality, because the thread which incremented open count past _min might get exception
```